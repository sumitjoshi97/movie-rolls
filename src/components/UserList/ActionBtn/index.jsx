import React, { Component } from 'react'

export default class ActionBtn extends Component {
  render() {
    const style =
      this.props.active === this.props.name
        ? { backgroundColor: '#333', color: '#fff' }
        : null

    return (
      <button
        name='movie'
        onClick={() => this.props.onclick(this.props.name)}
        style={style}
      >
        {this.props.text} {this.props.name === 'movie' ? 'movies' : 'shows'}
      </button>
    )
  }
}
