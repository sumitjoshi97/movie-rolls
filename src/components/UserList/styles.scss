.user-list {
  display: flex;
  flex-direction: column;

  &__action-btns {
    position: absolute;
    top: 10rem;
    width: 100%;
    display: flex;
    justify-content: flex-start;
    padding: 0 4rem;

    & > * {
      padding: 2rem 3rem;
      background: #212121;
      color: #d8d8d8;
      border: none;
      font-size: 1.6rem;
      letter-spacing: 1px;
      text-transform: uppercase;
      margin-right: 4rem;
      cursor: pointer;

      &:focus,
      &:active {
        outline: none;
      }
    }
  }

  &__items {
    padding-top: 20rem;
  }
}
