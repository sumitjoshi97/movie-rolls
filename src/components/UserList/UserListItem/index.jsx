import React from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'

import { setItemType } from '../../../actions'

import './styles.scss'

const UserListItem = ({ pageId, name, poster, type, ...props }) => {
  return (
    <div className='user-list-item'>
      <Link to={`/details/${pageId}`} onClick={() => props.setItemType(type)}>
        <img
          src={`https://image.tmdb.org/t/p/w185/${poster}`}
          alt={name}
          className='user-list-item__img'
        />
        <p className='user-list-item__name'>{name}</p>
      </Link>
    </div>
  )
}

export default connect(
  null,
  { setItemType }
)(UserListItem)
