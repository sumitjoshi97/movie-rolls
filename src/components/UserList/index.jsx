import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

import List from '../List'

import './styles.scss'

export default class UserList extends PureComponent {
  static propTypes = {
    list: PropTypes.object
  }

  renderItems = () => {
    const { list } = this.props

    let movieList = []
    let tvList = []

    if (list) {
      for (let item in list) {
        if (list[item].type === 'movie') {
          movieList.push(list[item])
        } else {
          tvList.push(list[item])
        }
      }

      return (
        <>
          <List name='movies' type='movie' items={movieList} />
          <List name='TV shows' type='tv' items={tvList} />
        </>
      )
    }
  }

  render() {
    return <div className='user-list'>{this.renderItems()}</div>
  }
}
