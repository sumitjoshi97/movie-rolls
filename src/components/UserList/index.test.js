import React from 'react'
import { shallow } from 'enzyme'

import UserList from './index'

describe('UserList component unit tests', () => {
  const list = {
    0: {
      type: 'movie',
      id: '01'
    },
    1: {
      type: 'tv',
      id: '02'
    }
  }

  const props = { list }

  const userListWrapper = shallow(<UserList {...props} />)

  test('should render properly', () => {
    expect(userListWrapper).toMatchSnapshot()
  })

  test('should have a movie list', () => {
    expect(userListWrapper.find({ items: [list[0]] })).toHaveLength(1)
  })

  test('should have a tv list', () => {
    expect(userListWrapper.find({ items: [list[1]] })).toHaveLength(1)
  })
})
