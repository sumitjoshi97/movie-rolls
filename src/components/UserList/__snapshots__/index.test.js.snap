// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList component unit tests should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserList
    list={
      Object {
        "0": Object {
          "id": "01",
          "type": "movie",
        },
        "1": Object {
          "id": "02",
          "type": "tv",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <React.Fragment>
        <List
          items={
            Array [
              Object {
                "id": "01",
                "type": "movie",
              },
            ]
          }
          name="movies"
          type="movie"
        />
        <List
          items={
            Array [
              Object {
                "id": "02",
                "type": "tv",
              },
            ]
          }
          name="TV shows"
          type="tv"
        />
      </React.Fragment>,
      "className": "user-list",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <List
            items={
              Array [
                Object {
                  "id": "01",
                  "type": "movie",
                },
              ]
            }
            name="movies"
            type="movie"
          />,
          <List
            items={
              Array [
                Object {
                  "id": "02",
                  "type": "tv",
                },
              ]
            }
            name="TV shows"
            type="tv"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "items": Array [
              Object {
                "id": "01",
                "type": "movie",
              },
            ],
            "name": "movies",
            "type": "movie",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "items": Array [
              Object {
                "id": "02",
                "type": "tv",
              },
            ],
            "name": "TV shows",
            "type": "tv",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <React.Fragment>
          <List
            items={
              Array [
                Object {
                  "id": "01",
                  "type": "movie",
                },
              ]
            }
            name="movies"
            type="movie"
          />
          <List
            items={
              Array [
                Object {
                  "id": "02",
                  "type": "tv",
                },
              ]
            }
            name="TV shows"
            type="tv"
          />
        </React.Fragment>,
        "className": "user-list",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <List
              items={
                Array [
                  Object {
                    "id": "01",
                    "type": "movie",
                  },
                ]
              }
              name="movies"
              type="movie"
            />,
            <List
              items={
                Array [
                  Object {
                    "id": "02",
                    "type": "tv",
                  },
                ]
              }
              name="TV shows"
              type="tv"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "items": Array [
                Object {
                  "id": "01",
                  "type": "movie",
                },
              ],
              "name": "movies",
              "type": "movie",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "items": Array [
                Object {
                  "id": "02",
                  "type": "tv",
                },
              ],
              "name": "TV shows",
              "type": "tv",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
